File: mobile/android/app/src/main/java/com/cdbl/leavemanager/data/api/AuthService.kt
package com.cdbl.leavemanager.data.api

import com.cdbl.leavemanager.data.model.LoginRequest
import com.cdbl.leavemanager.data.model.LoginResponse
import com.cdbl.leavemanager.data.model.VerifyOtpRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthService {
    @POST("auth/mobile-login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>

    @POST("auth/mobile-verify-otp")
    suspend fun verifyOtp(@Body request: VerifyOtpRequest): Response<LoginResponse>

    @retrofit2.http.GET("auth/me")
    suspend fun getCurrentUser(@retrofit2.http.Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.User>

    @retrofit2.http.POST("settings/change-password")
    suspend fun changePassword(
        @retrofit2.http.Header("Authorization") token: String,
        @retrofit2.http.Body request: com.cdbl.leavemanager.data.model.ChangePasswordRequest
    ): retrofit2.Response<com.cdbl.leavemanager.data.model.ChangePasswordResponse>

    @retrofit2.http.GET("user/profile")
    suspend fun getUserProfile(
        @retrofit2.http.Header("Authorization") token: String
    ): retrofit2.Response<com.cdbl.leavemanager.data.model.UserDetailsResponse>

    @retrofit2.http.PUT("user/profile")
    suspend fun updateProfile(
        @retrofit2.http.Header("Authorization") token: String,
        @retrofit2.http.Body request: com.cdbl.leavemanager.data.model.UpdateProfileRequest
    ): retrofit2.Response<Map<String, Any>>
}

File: mobile/ios/CDBLLeaveManager/CDBLLeaveManager/Core/Networking/AuthService.swift
//
//  AuthService.swift
//  CDBLLeaveManager
//
//  Authentication API service.
//

import Foundation

actor AuthService {
    static let shared = AuthService()
    
    private let client = APIClient.shared
    
    private init() {}
    
    // MARK: - Login
    
    func login(email: String, password: String, skipOtp: Bool = false) async throws -> LoginResponse {
        let request = LoginRequest(email: email, password: password, skipOtp: skipOtp)
        return try await client.request(
            "auth/mobile-login",
            method: .post,
            body: request,
            requiresAuth: false
        )
    }
    
    // MARK: - OTP Verification
    
    func verifyOtp(email: String, code: String) async throws -> LoginResponse {
        let request = VerifyOtpRequest(email: email, code: code)
        return try await client.request(
            "auth/mobile-verify-otp",
            method: .post,
            body: request,
            requiresAuth: false
        )
    }
    
    // MARK: - Get Current User
    
    func getCurrentUser() async throws -> AuthUser {
        return try await client.request("auth/me")
    }
    
    // MARK: - Get User Profile
    
    func getUserProfile() async throws -> UserDetailsResponse {
        return try await client.request("user/profile")
    }
    
    // MARK: - Update Profile
    
    func updateProfile(_ request: UpdateProfileRequest) async throws -> APIResponse<UserDetailsResponse> {
        return try await client.request(
            "user/profile",
            method: .put,
            body: request
        )
    }
    
    // MARK: - Change Password
    
    func changePassword(_ request: ChangePasswordRequest) async throws -> ChangePasswordResponse {
        return try await client.request(
            "settings/change-password",
            method: .post,
            body: request
        )
    }
    
    // MARK: - Logout
    
    func logout() async {
        TokenManager.shared.clearToken()
    }
}
